src/db/node.rs

```diff
@@ -2,6 +2,45 @@ use std::collections::VecDeque;
 
 use crate::db::{entry::Entry, group::Group};
 
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub(crate) enum NodePathElement<'a> {
+    #[allow(dead_code)]
+    UUID(&'a str),
+    Title(&'a str),
+}
+
+pub(crate) type NodePath<'a> = Vec<NodePathElement<'a>>;
+
+impl<'a> NodePathElement<'_> {
+    pub(crate) fn matches(&self, node: &Node) -> bool {
+        let uuid = match node {
+            Node::Entry(e) => e.uuid,
+            Node::Group(g) => g.uuid,
+        };
+        let title = match node {
+            Node::Entry(e) => e.get_title(),
+            Node::Group(g) => Some(g.name.as_ref()),
```

Comment:
There's no `get_name()` for a group?

Reply:
Not yet, but I opened [a PR](https://github.com/sseemayer/keepass-rs/pull/189) to add it.